dist/common/scripts/scylla_util.py

```diff
@@ -343,6 +343,183 @@ def user_data(self):
         except urllib.error.HTTPError:  # empty user-data
             return ""
 
+class azure_instance:
+    """Describe several aspects of the current Azure instance"""
+
+    EPHEMERAL = "ephemeral"
+    ROOT = "root"
+    GETTING_STARTED_URL = "http://www.scylladb.com/doc/getting-started-azure/"
+    META_DATA_BASE_URL = "http://169.254.169.254/1.0/meta-data/instance/"
+    ENDPOINT_SNITCH = "GossipingPropertyFileSnitch"
+
+    def __init__(self):
+        self.__type = None
+        self.__cpu = None
+        self.__memoryGB = None
+        self.__nvmeDiskCount = None
+        self.__firstNvmeSize = None
+        self.__osDisks = None
+
+    @staticmethod
+    def is_azure_instance():
```

Comment:
looks like a copy-paste from `is_gce_instance` with small changes, generalize by moving the code into separate function or create a base class that all `*_instance` classes will inherit from.

Reply:
refactoring will happen in next PR